exporter/containerimage/attestations.go

```diff
@@ -1,10 +1,169 @@
 package containerimage
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+
+	intoto "github.com/in-toto/in-toto-golang/in_toto"
+	"github.com/moby/buildkit/cache"
+	"github.com/moby/buildkit/exporter/attestation"
+	gatewaypb "github.com/moby/buildkit/frontend/gateway/pb"
+	"github.com/moby/buildkit/session"
+	"github.com/moby/buildkit/solver"
+	"github.com/moby/buildkit/solver/result"
 	ocispecs "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/pkg/errors"
+	"github.com/spdx/tools-golang/jsonloader"
+	"github.com/spdx/tools-golang/jsonsaver"
+	"github.com/spdx/tools-golang/spdx"
 )
 
 var intotoPlatform ocispecs.Platform = ocispecs.Platform{
 	Architecture: "unknown",
 	OS:           "unknown",
 }
+
+// supplementSBOM modifies SPDX attestations to include the file layers
+func supplementSBOM(ctx context.Context, s session.Group, target cache.ImmutableRef, targetRemote *solver.Remote, refs map[string]cache.ImmutableRef, att result.Attestation) (result.Attestation, error) {
+	if att.Kind != gatewaypb.AttestationKindInToto {
+		return att, nil
+	}
+	if att.InToto.PredicateType != intoto.PredicateSPDX {
+		return att, nil
+	}
+
+	content, err := attestation.Read(ctx, s, refs, att)
+	if err != nil {
+		return att, err
+	}
+
+	doc, err := decodeSPDX(content)
+	if err != nil {
+		return att, err
+	}
+
+	layers := newFileLayerFinder(target, targetRemote)
+	modifyFile := func(f *spdx.File2_2) error {
+		_, desc, err := layers.find(ctx, s, f.FileName)
+		if err != nil {
+			return err
+		}
+		f.FileComment = fmt.Sprintf("layerID: %s", desc.Digest.String())
+		return nil
+	}
+	for _, f := range doc.UnpackagedFiles {
+		if err := modifyFile(f); err != nil {
+			return att, err
+		}
+	}
+	for _, p := range doc.Packages {
+		for _, f := range p.Files {
+			if err := modifyFile(f); err != nil {
+				return att, err
+			}
+		}
+	}
+
+	dt, err := encodeSPDX(doc)
```

Comment:
If we're modifying the SPDX result, should we append Buildkit to the [creators field](https://spdx.github.io/spdx-spec/v2.3/document-creation-information/#68-creator-field)?